### PR TITLE
feat: migrate `Redirection` component to composition api

### PR DESCRIPTION
**Pull request by lauramargar**
# Pull request template
 
Migrate `Redirection` component to composition API

## Motivation and context

- [ ] Dependencies. If any, specify:
- [X] Open issue. If applicable, link: [EMP-4334](https://searchbroker.atlassian.net/browse/EMP-4334)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

[EMP-4334]: https://searchbroker.atlassian.net/browse/EMP-4334?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/x-components/src/x-modules/search/components/redirection.vue [hunk 1]
```diff
@@ -1,17 +1,16 @@
 <template>
-  <div v-if="redirection && $scopedSlots.default" class="x-redirection" data-test="redirection">
+  <div v-if="redirection && slots.default" class="x-redirection" data-test="redirection">
```
Comment (victorcg88) [inline packages/x-components/src/x-modules/search/components/redirection.vue:2]: Not sure if this `slots.default` would work as you expect here. As slots management has changed and `slots.defaut `is now a function.

File: packages/x-components/src/x-modules/search/components/redirection.vue [hunk 2]
```diff
@@ -20,103 +19,125 @@
...
+       * @public
+       */
+      delayInSeconds: {
+        type: Number,
+        default: 0
       }
-    }
+    },
+    setup(props, { slots }) {
+      const $x = use$x();
```
Comment (victorcg88) [inline packages/x-components/src/x-modules/search/components/redirection.vue:48]: Could you reorder the code into blocks of const, refs, watchers, and functions? With these patterns, we improve the developer experience as we can go into any component knowing its distribution beforehand.

File: packages/_vue3-migration-test/src/x-modules/search/components/test-redirection.vue [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<template>
```
Comment (victorcg88) [inline packages/_vue3-migration-test/src/x-modules/search/components/test-redirection.vue:1]: I can't make this feature work on my local machine, as the search request is not executed.

File: packages/x-components/src/x-modules/search/components/__tests__/redirection.spec.ts [hunk 4]
```diff
@@ -150,19 +153,17 @@ describe('testing Redirection component', () => {
...
   //eslint-disable-next-line max-len
   it('redirects and emits the `UserClickedARedirection` event in auto mode and 0 seconds of delay', () => {
-    const { wrapper } = renderRedirection({ delayInSeconds: 0 });
-    const onUserClickedARedirection = jest.fn();
+    const { onUserClickedARedirection } = renderRedirection({ delayInSeconds: 0 });
 
-    wrapper.vm.$x.on('UserClickedARedirection', true).subscribe(onUserClickedARedirection);
     // The delay 0 runs so fast the we need to force the test to simulate a wait.
-    jest.advanceTimersByTime(1);
+    jest.advanceTimersByTime(0);
```
[the remark this answers is not in the document]
Comment (lauramargar) [reply packages/x-components/src/x-modules/search/components/__tests__/redirection.spec.ts:159]: No, I forgot to rollback before committing changes

File: packages/_vue3-migration-test/src/x-modules/search/components/test-redirection.vue (hunk 3 above)
Comment (lauramargar) [reply packages/_vue3-migration-test/src/x-modules/search/components/test-redirection.vue:1]: In my local machine is working. What Vuex version do you have? Take into account this comment of @joseacabaneros https://github.com/empathyco/x/pull/1497#discussion_r1635139691

Comment (victorcg88) [reply packages/_vue3-migration-test/src/x-modules/search/components/test-redirection.vue:1]: Sorry, yes it was it. Now it works, thanks.

**Review by victorcg88 — APPROVED** [review]

**Pull request merged**